### ISSUE TITLE
App service failed TLS handshake failed

### ISSUE THREAD

**Issue by romaric1** [gravitational/teleport#23376]
Expected behavior:
![image](https://user-images.githubusercontent.com/31241280/226637647-d7c9903f-1ffc-4f1d-ac61-9fded35c1bb4.png)

Current behavior:

![image](https://user-images.githubusercontent.com/31241280/226637772-1c346dd9-912c-4dad-aa19-e868b39e936d.png)

Bug details:
- Teleport version : 12.x ? same version of this documentation -> https://goteleport.com/docs/try-out-teleport/local-kubernetes/
- Recreation steps : 
I follow this getting started -> https://goteleport.com/docs/try-out-teleport/local-kubernetes/

- Debug logs : 
2023-03-21T13:47:25Z WARN [APP:SERVI] Failed to handle client connection. error:[
ERROR REPORT:
Original Error: *errors.errorString EOF
Stack Trace:
	github.com/gravitational/teleport/lib/srv/app/server.go:883 github.com/gravitational/teleport/lib/srv/app.(*Server).getConnectionInfo
	github.com/gravitational/teleport/lib/srv/app/server.go:674 github.com/gravitational/teleport/lib/srv/app.(*Server).handleConnection
	github.com/gravitational/teleport/lib/srv/app/server.go:641 github.com/gravitational/teleport/lib/srv/app.(*Server).HandleConnection
	github.com/gravitational/teleport/lib/reversetunnel/transport.go:275 github.com/gravitational/teleport/lib/reversetunnel.(*transport).start
	github.com/gravitational/teleport/lib/reversetunnel/agent.go:581 github.com/gravitational/teleport/lib/reversetunnel.(*agent).handleDrainChannels.func2
	runtime/asm_amd64.s:1594 runtime.goexit
User Message: TLS handshake failed
	EOF] app/server.go:648

Important :
- I tried with --version=11.0.0 of the teleport-kube-agent

- At the step to use minikube tunnel there are a github issue so i make a port- forward kubectl port-forward svc/teleport-cluster 4443:443 -n teleport-cluster -> so I don't have external IP for my service

- my minikube node and my computer host  are not a the same time : 
ex node : Tue 2023-03-21 14:31:03 UTC vs computer host mar. 2023-03-21 15:31:14 CET

Thanks in advance for the help

**Comment by HagaiBarel** [comment]
Seeing the same behaviour, altough the setup is a bit different. In our case:
* Teleport cluster (proxy/auth) was upgraded from 11.3.6 -> 12.1.1 (OSS version)
* Agents connected to the cluster are running version 11.3.6

Application access doesn't work when trying to launch the app from the teleport UI, but does work when connecting via `tsh proxy app <app-name>`. Same error is the OP mentions.

**Comment by benarent** [comment]
We recently debugged this issue. The reason for this error message is due to DNS, it fails `User Message: TLS handshake failed` because it's looking to go to '*.teleport.TLD.com` and we had only set up `teleport.TLD.com` and not the wildcard subdomain. 

See the below two DNS items. If using ACME we'll obtain a wild card cert for you. 

<img width="501" alt="image" src="https://user-images.githubusercontent.com/559288/228346110-86ae1501-0bae-4f40-bd75-7d6f1b9b9ba6.png">
<img width="446" alt="image" src="https://user-images.githubusercontent.com/559288/228346129-8efa5a60-2cd0-40f7-bd15-657f8c7a85e7.png">
